src/vm/stubgen.cpp

```diff
@@ -2252,7 +2252,22 @@ ILStubLinker::ILStubLinker(Module* pStubSigModule, const Signature &signature, S
         if (uStubCallingConvInfo & IMAGE_CEE_CS_CALLCONV_GENERIC)
             IfFailThrow(m_managedSigPtr.GetData(NULL));    // skip number of type parameters
 
-        IfFailThrow(m_managedSigPtr.GetData(NULL));        // skip number of parameters
+        if (fIsReverseStub)
+        {
+            ULONG numParams = 0;
+            IfFailThrow(m_managedSigPtr.GetData(&numParams));
+            // As per ECMA 335, the max number of parameters is 0x1FFFFFFF (see section 11.23.2), which fits in a 32-bit signed integer 
+            // So this cast is safe.
+            m_iTargetStackDelta -= (INT)numParams;
+            if (!m_StubHasVoidReturnType)
+            {
+                m_iTargetStackDelta++;
+            }
+        }
+        else
+        {
+            IfFailThrow(m_managedSigPtr.GetData(NULL));        // skip number of parameters
```

Comment:
Instead of this, I would pull this out above the `if` and only use the result if `fIsReverseStub` is true.